app/controllers/foreman_tasks/api/tasks_controller.rb

```diff
@@ -122,6 +121,42 @@ def bulk_resume
         }
       end
 
+      api :POST, '/tasks/bulk_cancel', N_('Cancel all cancellable tasks')
+      param :search, String, :desc => N_('Cancel tasks matching search string')
+      param :task_ids, Array, :desc => N_('Cancel specific tasks by ID')
+      def bulk_cancel
+        scope = resource_scope
+        scope = scope.search_for(params[:search]) if params[:search]
+        scope = scope.select('DISTINCT foreman_tasks_tasks.*')
+        if params[:search].nil? && params[:task_ids].nil?
+          raise ArgumentError, _('Please provide at least one of search or task_ids parameters in the request')
+        end
```

Comment:
I would move this to the top of the method

Reply:
Moved this and the second one up